fastchat/serve/gradio_block_arena_vision_anony.py

```diff
@@ -242,6 +246,71 @@ def clear_history(request: gr.Request):
         + [""]
     )
 
+def extract_text_from_pdf(pdf_file_path):
+    """Extract text from a PDF file."""
+    try:
+        with open(pdf_file_path, 'rb') as f:
+            reader = PyPDF2.PdfReader(f)
+            pdf_text = ""
+            for page in reader.pages:
+                pdf_text += page.extract_text()
+            return pdf_text
+    except Exception as e:
+        logger.error(f"Failed to extract text from PDF: {e}")
+        return None
+    
+import os
+import nest_asyncio
+from llama_parse import LlamaParse
+
+nest_asyncio.apply()  # Ensure compatibility with async environments
```

Comment:
why do we need this?

Reply:
I'm not exactly sure where it is being used but it was part of the script for LlamaParse API calls.